### PR TITLE
Resource optimisation

### PR DESCRIPTION
**Pull request by jfy133**
Based on DeepEvo big test (tiny/crappy samples to modern relatively-large monster assemblies)

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/base.config [hunk 1]
```diff
@@ -62,4 +62,148 @@ process {
...
+    withName:ANTISMASH_ANTISMASHLITE {
+        errorStrategy = { task.exitStatus in [1] ? 'ignore' : task.exitStatus in [143,137,104,134,139] ? 'retry' : 'finish' }
+    }
+
+    /*
+    * Resource specifications
+    */
+    withName: GUNZIP {
+        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:74]:
```suggestion
        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 2]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: UNTAR {
+        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:79]:
```suggestion
        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 3]
```diff
@@ -62,4 +62,148 @@ process {
...
+
+    // TODO not tested
+    withName: PROKKA {
+        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
+        cpus   = { check_max( 4    * task.attempt, 'cpus'   ) }
+    }
+
+    withName: PRODIGAL_GFF {
+        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:90]:
```suggestion
        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 4]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: PRODIGAL_GBK {
+        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:95]:
```suggestion
        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 5]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: ABRICATE_RUN {
+        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
+        cpus   = { check_max( 4    * task.attempt, 'cpus'   ) }
+    }
+
+    withName: AMRFINDERPLUS_RUN {
+        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:105]:
```suggestion
        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 6]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: DEEPARG_DOWNLOADDATA {
+        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:110]:
```suggestion
        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 7]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: DEEPARG_PREDICT {
+        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:115]:
```suggestion
        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 8]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: RGI_MAIN {
+        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
+        cpus   = { check_max( 4    * task.attempt, 'cpus'   ) }
+    }
+
+    withName: AMPIR {
+        memory = { check_max( 8.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:130]:
```suggestion
        memory = { check_max( 8.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 9]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: AMPLIFY_PREDICT {
+        memory = { check_max( 16.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:135]:
```suggestion
        memory = { check_max( 16.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 10]
```diff
@@ -62,4 +62,148 @@ process {
...
+
+    withName: ANTISMASH_ANTISMASHLITE {
+        memory = { check_max( 64.GB * task.attempt, 'memory'  ) }
+        cpus   = { check_max( 8    * task.attempt, 'cpus'   ) }
+        time   = { check_max( 12.h  * task.attempt, 'time'   ) }
+    }
+
+    withName: ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES {
+        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:161]:
```suggestion
        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 11]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: DEEPBGC_DOWNLOAD {
+        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:166]:
```suggestion
        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 12]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: DEEPBGC_PIPELINE {
+        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:171]:
```suggestion
        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 13]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: GECCO_RUN {
+        memory = { check_max( 8.GB * task.attempt, 'memory'  ) }
+        cpus   = { check_max( 4    * task.attempt, 'cpus'   ) }
+    }
+
+    withName: HAMRONIZATION_ABRICATE {
+        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:182]:
```suggestion
        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 14]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: HAMRONIZATION_AMRFINDERPLUS {
+        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:187]:
```suggestion
        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 15]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: HAMRONIZATION_DEEPARG {
+        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:192]:
```suggestion
        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 16]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: HAMRONIZATION_RGI {
+        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:197]:
```suggestion
        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 17]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: HAMRONIZATION_FARGENE {
+        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:202]:
```suggestion
        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 18]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: HAMRONIZATION_SUMMARIZE {
+        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
+        cpus = 1
```
Comment (jasmezz) [inline conf/base.config:207]:
```suggestion
        memory = { check_max( 4.GB * task.attempt, 'memory'  ) }
        cpus   = 1
```

File: conf/base.config [hunk 19]
```diff
@@ -62,4 +62,148 @@ process {
...
+    }
+
+    withName: BGC_HMMER_HMMSEARCH {
+        memory = { check_max( 1.GB * task.attempt, 'memory'  ) }
+        cpus   = { check_max( 4    * task.attempt, 'cpus'   ) }
+    }
+
+    withName: ANTISMASH_ANTISMASHLITE {
+        memory = { check_max( 64.GB * task.attempt, 'memory'  ) }
+        cpus   = { check_max( 8    * task.attempt, 'cpus'   ) }
```
Comment (jasmezz) [inline conf/base.config:155]:
```suggestion
        memory = { check_max( 64.GB * task.attempt, 'memory'  ) }
        cpus   = { check_max( 8     * task.attempt, 'cpus'   ) }
```

**Review by jasmezz — APPROVED** [review]

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Pull request merged**